usl_models/usl_models/flood_ml/prediction_dataset.py

```diff
@@ -0,0 +1,107 @@
+"""tf.data.Datasets for prediction using FloodML model trained on CityCAT data."""
+
+import logging
+from typing import Dict, Iterator, Optional, TypedDict
+
+from google.cloud import storage  # type:ignore[attr-defined]
+import numpy
+import tensorflow as tf
+
+from usl_models.flood_ml import constants, dataset
+
+
+class PredictionInput(TypedDict):
+    """Input tensors dictionary used for prediction."""
+
+    geospatial: tf.Tensor
+    temporal: tf.Tensor
+    spatiotemporal: tf.Tensor
+    metadata: tf.Tensor
```

Comment:
I'd just add metadata to the existing `FlodModel.Input` dict. Maybe make it an Optional field. If training ignores the metadata we supply, that's fine. Doesn't hurt to add it since we need to look up the metadata anyway.

I think that would help share more code between the two dataset loading code paths.